src/scripts/underground/Mine.ts

```diff
@@ -190,6 +190,10 @@ class Mine {
             } else {
                 switch (reward.valueType) {
                     case 'Diamond': {
+                        if (reward.isShard()) {
+                            res.shards++;
+                            break;
+                        }
```

Comment:
Not really related to this PR, but shouldn't the `valueType` for shards not be `Diamond`?
Would rather us sort that out first before we engrain the shards more into the `Diamond` type.